h5py/_hl/group.py

```diff
@@ -206,9 +222,7 @@ def create_dataset_like(self, name, other, **kwupdate):
         shape and dtype, in which case the provided values take precedence over
         those from `other`.
         """
-        for k in ('shape', 'dtype', 'chunks', 'compression',
-                  'compression_opts', 'scaleoffset', 'shuffle', 'fletcher32',
-                  'fillvalue'):
+        for k in ('shape', 'dtype'):
```

Comment:
This is starting to be scope-creep, but adding a `dcpl.to_dict()` method?

Reply:
The problem with `track_order` is that it maps to both HDF5 attr creation order and HDF5 links creation order, and collapsing the latter two to a single value while copying will lose information. So `dcpl.to_dict()` is not flawless.
What about extracting non-public `make_dsid()` similar to `make_fid()`?

Reply:
Will do!